app/src/main/java/com/android/ootd/model/posts/LikesFirestoreRepository.kt

```diff
@@ -0,0 +1,92 @@
+package com.android.ootd.model.posts
+
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.SetOptions
+import kotlinx.coroutines.tasks.await
+
+const val LIKE_COLLECTION_PATH = "likes"
+
+class LikesFirestoreRepository(private val db: FirebaseFirestore) : LikesRepository {
+
+  override suspend fun likePost(like: Like) {
+    try {
+      db.collection(LIKE_COLLECTION_PATH)
+          .document(like.postId)
+          .collection("users")
+          .document(like.likerUserId)
+          .set(
+              mapOf(
+                  "postId" to like.postId,
+                  "likerUserId" to like.likerUserId,
+                  "timestamp" to like.timestamp),
+              SetOptions.merge())
+          .await()
+    } catch (e: Exception) {
+      Log.e("LikeRepository", "Failed to add like for post ${like.postId}", e)
+      throw e
+    }
+  }
+
+  override suspend fun unlikePost(postId: String, userId: String) {
+    try {
+      db.collection(LIKE_COLLECTION_PATH)
+          .document(postId)
+          .collection("users")
+          .document(userId)
+          .delete()
+          .await()
+    } catch (e: Exception) {
+      Log.e("LikeRepository", "Failed to remove like for post $postId", e)
```

Comment:
Since, you are using `LikeRepository` as a tag in different functions. Instead of repeating over and over again, what do you think about doing them as 


```suggestion
      private const val TAG_LIKE_REPOSITORY = LikeRepository
```